File: backend/hackathon/src/main/java/com/dev_trails/hackathon/ReferralCode.java
package com.dev_trails.hackathon;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Column;

@Entity
public class ReferralCode {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(unique = true)
    public String code;
    public Long ownerRiderId;
    public String ownerName;
    public Integer timesUsed;
    public Double rewardEarned;
    public String createdAt;
}

File: backend/models/ReferralCode.js
const mongoose = require('mongoose');

const ReferralCodeSchema = new mongoose.Schema({
  code: { type: String, unique: true, sparse: true },
  owner_rider_id: { type: mongoose.Schema.Types.Mixed },
  owner_name: { type: String },
  times_used: { type: Number, default: 0 },
  reward_earned: { type: Number, default: 0 },
  created_at: { type: String },
}, { timestamps: false });

ReferralCodeSchema.set('toJSON', { virtuals: true });
ReferralCodeSchema.virtual('id').get(function () { return this._id.toHexString(); });

module.exports = mongoose.model('ReferralCode', ReferralCodeSchema);
